crates/proof-of-sql/src/sql/proof_plans/table_exec.rs

```diff
@@ -0,0 +1,126 @@
+use crate::{
+    base::{
+        commitment::Commitment,
+        database::{
+            Column, ColumnField, ColumnRef, CommitmentAccessor, DataAccessor, MetadataAccessor,
+            OwnedTable, TableRef,
+        },
+        map::IndexSet,
+        proof::ProofError,
+        scalar::Scalar,
+    },
+    sql::proof::{
+        CountBuilder, FinalRoundBuilder, FirstRoundBuilder, ProofPlan, ProverEvaluate,
+        VerificationBuilder,
+    },
+};
+use alloc::vec::Vec;
+use bumpalo::Bump;
+use core::iter::repeat_with;
+use serde::{Deserialize, Serialize};
+
+/// Source [`ProofPlan`] for (sub)queries with table source such as `SELECT col from tab;`
+/// Inspired by `DataFusion` data source [`ExecutionPlan`]s such as [`ArrowExec`] and [`CsvExec`].
+#[derive(Debug, PartialEq, Serialize, Deserialize)]
+pub struct TableExec {
+    /// Table reference
+    pub table_ref: TableRef,
+    /// Schema of the table
+    pub schema: Vec<ColumnField>,
+}
+
+impl TableExec {
+    /// Creates a new [`TableExec`].
+    #[must_use]
+    pub fn new(table_ref: TableRef, schema: Vec<ColumnField>) -> Self {
+        Self { table_ref, schema }
+    }
+
+    /// Returns the entire table.
+    fn get_table<'a, S: Scalar>(&self, accessor: &'a dyn DataAccessor<S>) -> Vec<Column<'a, S>> {
+        self.schema
+            .iter()
+            .map(|field| {
+                accessor.get_column(ColumnRef::new(
+                    self.table_ref,
+                    field.name(),
+                    field.data_type(),
+                ))
+            })
+            .collect()
+    }
+}
+
+impl ProofPlan for TableExec {
+    fn count(
+        &self,
+        _builder: &mut CountBuilder,
+        _accessor: &dyn MetadataAccessor,
+    ) -> Result<(), ProofError> {
+        Ok(())
+    }
+
+    fn get_length(&self, accessor: &dyn MetadataAccessor) -> usize {
+        accessor.get_length(self.table_ref)
+    }
+
+    fn get_offset(&self, accessor: &dyn MetadataAccessor) -> usize {
+        accessor.get_offset(self.table_ref)
+    }
+
+    #[allow(unused_variables)]
+    fn verifier_evaluate<C: Commitment>(
+        &self,
+        builder: &mut VerificationBuilder<C>,
+        _accessor: &dyn CommitmentAccessor<C>,
+        _result: Option<&OwnedTable<C::Scalar>>,
+    ) -> Result<Vec<C::Scalar>, ProofError> {
+        Ok(repeat_with(|| builder.consume_intermediate_mle())
+            .take(self.schema.len())
+            .collect::<Vec<_>>())
+    }
+
+    fn get_column_result_fields(&self) -> Vec<ColumnField> {
+        self.schema.clone()
+    }
+
+    fn get_column_references(&self) -> IndexSet<ColumnRef> {
+        self.schema
+            .iter()
+            .map(|field| ColumnRef::new(self.table_ref, field.name(), field.data_type()))
+            .collect()
+    }
+
+    fn get_table_references(&self) -> IndexSet<TableRef> {
+        core::iter::once(self.table_ref).collect()
```

Comment:
NIT:
```suggestion
        indexset! {self.table.table_ref}
```

Would require adding thing
```rust
/// Create an [`IndexSet`][self::IndexSet] from a list of values
macro_rules! indexset_macro {
    ($($value:expr,)+) => { $crate::base::map::indexset!($($value),+) };
    ($($value:expr),*) => {
        {
            const CAP: usize = <[()]>::len(&[$({ stringify!($value); }),*]);
            let mut set = $crate::base::map::IndexSet::with_capacity_and_hasher(CAP, <_>::default());
            $(
                set.insert($value);
            )*
            set
        }
    };
}

pub(crate) use indexset_macro as indexset;
```